core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -772,6 +772,15 @@ final class ZManaged[-R, +E, +A] private (val zio: ZIO[(R, ZManaged.ReleaseMap),
         .applyOrElse[A, ZManaged[R1, E1, A]](v, ZManaged.succeedNow)
     }
 
+  /**
+   * Runs all the finalizers associated with this scope. This is useful to
+   * conceptually "close" a scope when composing multiple managed effects.
+   * Note that this is only safe if the result of this managed effect is valid
+   * outside its scope.
+   */
+  def release: ZManaged[R, E, A] =
```

Comment:
I think there's an issue here in that `a *> b.release *> c` won't work, as after `b.release`, the ReleaseMap is already closed.

The workaround may be for this combinator to run `self` on a nested `ReleaseMap`.

Reply:
Added a test for this. Seems to be working as expected:

```scala
testM("closes the scope") {
  val expected = Chunk("acquiring a", "acquiring b", "releasing b", "acquiring c", "releasing c", "releasing a")
  for {
    ref     <- Ref.make[Chunk[String]](Chunk.empty)
    a       = Managed.make(ref.update(_ + "acquiring a"))(_ => ref.update(_ + "releasing a"))
    b       = Managed.make(ref.update(_ + "acquiring b"))(_ => ref.update(_ + "releasing b"))
    c       = Managed.make(ref.update(_ + "acquiring c"))(_ => ref.update(_ + "releasing c"))
    managed = a *> b.release *> c
    _       <- managed.useNow
    log     <- ref.get
  } yield assert(log)(equalTo(expected))
}
```